test/e2e/conflict_remote_namespace_e2e/conflict_creation_test.go

```diff
@@ -0,0 +1,122 @@
+	// local name of the test namespace.
+	testNamespaceName = "test-namespace-conflict"
+)
+
+var _ = Describe("Liqo E2E", func() {
+	var (
+		ctx         = context.Background()
+		testContext = tester.GetTester(ctx)
+		interval    = 1 * time.Second
+		timeout     = 10 * time.Second
+		// longTimeout is used in situations that may take longer to be performed
+		longTimeout = 2 * time.Minute
+		localIndex  = 0
+		// index of the cluster on which a remote namespace with the same name already exists.
+		remoteIndex             = 2
+		localClusterID          = testContext.Clusters[localIndex].ClusterID
+		remoteTestNamespaceName = fmt.Sprintf("%s-%s", testNamespaceName, localClusterID)
+	)
+
+	Context(fmt.Sprintf("Create a namespace inside the cluster '%d' and check what happen if a remote namespaace in the cluster "+
+		"'%d' already exists. Remove the label and check the deletion of the remote namespaces.", localIndex, remoteIndex), func() {
+
+		It(fmt.Sprintf("Create a namespace inside the cluster '%d' and check what happen "+
+			"if a remote namespaace in the cluster '%d' already exists.", localIndex, remoteIndex), func() {
+			namespace := &corev1.Namespace{}
+			namespaceOffloading := &offloadingv1alpha1.NamespaceOffloading{}
+
+			By(fmt.Sprintf(" 1 - Creating the remote namespace inside the cluster '%d'", remoteIndex))
+			Eventually(func() error {
+				return testutils.CreateNamespaceWithoutNamespaceOffloading(ctx, testContext.Clusters[remoteIndex].ControllerClient, remoteTestNamespaceName)
+			}, timeout, interval).Should(Succeed())
+
+			By(fmt.Sprintf(" 2 - Creating the local namespace inside the cluster '%d'", localIndex))
+			Eventually(func() error {
+				return testutils.CreateNamespaceWithNamespaceOffloading(ctx, testContext.Clusters[localIndex].ControllerClient, testNamespaceName)
+			}, timeout, interval).Should(Succeed())
+
+			By(" 3 - Getting the NamespaceOffloading resource")
+			Eventually(func() error {
+				if err := testContext.Clusters[localIndex].ControllerClient.Get(ctx,
+					types.NamespacedName{Namespace: testNamespaceName, Name: liqoconst.DefaultNamespaceOffloadingName}, namespaceOffloading); err != nil {
+					return err
+				}
+				if namespaceOffloading.Status.OffloadingPhase != offloadingv1alpha1.SomeFailedOffloadingPhaseType {
+					return fmt.Errorf("the NamespaceOffloading resource has the wrong OffloadingPhase: %s",
+						namespaceOffloading.Status.OffloadingPhase)
+				}
+				return nil
+			}, longTimeout, interval).Should(Succeed())
+
+			By(fmt.Sprintf(" 4 - Deleting the remote namespace inside the cluster '%d'", remoteIndex))
+			Eventually(func() metav1.StatusReason {
+				err := testContext.Clusters[remoteIndex].ControllerClient.Get(ctx,
+					types.NamespacedName{Name: remoteTestNamespaceName}, namespace)
+				_ = testContext.Clusters[remoteIndex].ControllerClient.Delete(ctx, namespace)
+				return apierrors.ReasonForError(err)
+			}, timeout, interval).Should(Equal(metav1.StatusReasonNotFound))
```

Comment:
Here you risk racing against your controller. You delete the namespace, then check for its actual removal, but it might already have been recreated.

Reply:
yes you are right, but in the particular scenario of this test the controller should resync every 30 sec, so there shouldn't be a race condition with the controller. I understand that changing the resync timing could be a problem, I will try a different solution even if the behavior to test is just this, thank you

Reply:
I would propose the following (let me know if sounds ok to you):
1. Get the namespace, and save the `uid`
2. Delete the namespace, without waiting for its removal and retrying only in case of error
3. Assert that eventually the namespace still exists, but has a different `uid` (hence, it must have been recreated)

Reply:
Great solution, thank you for the advise :1st_place_medal: